packages/tenant-process/src/services/tenantService.ts

```diff
@@ -1153,16 +1153,112 @@ export function tenantServiceBuilder(
       logger.info(`Retrieving Tenant with Selfcare Id ${selfcareId}`);
       return readModelService.getTenantBySelfcareId(selfcareId);
     },
+
+    async internalUpsertTenant(
+      internalTenantSeed: tenantApi.InternalTenantSeed,
+      { correlationId, logger }: WithLogger<AppContext>
+    ): Promise<Tenant> {
+      logger.info(
+        `Updating tenant with external id ${internalTenantSeed.externalId} via internal request`
+      );
+
+      const existingTenant = await readModelService.getTenantByExternalId(
+        internalTenantSeed.externalId
+      );
+
+      assertTenantExists(
+        unsafeBrandId(
+          `${internalTenantSeed.externalId.origin}/${internalTenantSeed.externalId.value}`
+        ),
+        existingTenant
+      );
+
+      const attributesExternalIds = internalTenantSeed.certifiedAttributes.map(
+        (externalId) =>
+          ExternalId.parse({
+            value: externalId.code,
+            origin: externalId.origin,
+          })
+      );
+
+      const attributesByExternalId =
+        await readModelService.getAttributesByExternalIds(
+          attributesExternalIds
+        );
+
+      for (const attributeToAssign of attributesExternalIds) {
+        if (
+          !attributesByExternalId.find(
```

Comment:
```suggestion
      attributesExternalIds.forEach((attributeToAssign) => {
        if (
          !attributesByExternalId.some(
```
I would use a `forEach` and `some` instead of `find`